### PR TITLE
feat(tianmu):support the ALTER syntax, ALTER table t1 rename t2, add c char(10) (#760)

### LINKED ISSUE #760: feature: Dose not  support the ALTER syntax, ALTER table t1 rename t2, add c char(10);
**Describe the solution you'd like**

Suggest to support the ALTER sytax : 
ALTER table t1 rename t2, add c char(10);

**Additional context**

 The current situation is as follows：
```shell
mysql> create table t1 (i int primary key);
Query OK, 0 rows affected (0.00 sec)

mysql> alter table t1 rename t2, add c char(10);
ERROR 2233 (HY000): Be disgraceful to storage engine, operating is forbidden!
```

### PR DESCRIPTION
**Pull request by DandreChen**
## Summary about this PR

Issue Number: close #760 

[summary]
fix some warnings
add mtr for alter table
support alter table: mix add/drop column、order column and other syntaxs to use

## Tests Check List

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [x] New Feature
- [ ] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/core/temp_table.h [hunk 1]
```diff
@@ -111,12 +111,12 @@ class TempTable : public JustATable {
                                                                       // values (incl. null, if flag set)
     uint64_t ExactDistinctVals([[maybe_unused]] Filter *f) override { return common::NULL_VALUE_64; }
     bool IsDistinct([[maybe_unused]] Filter *f) override { return false; }
-    size_t MaxStringSize(Filter *f = nullptr) override;  // maximal byte string length in column
-    int64_t RoughMin(Filter *f = nullptr, common::RSValue *rf = nullptr) override {
+    size_t MaxStringSize([[maybe_unused]] Filter *f = nullptr) override;  // maximal byte string length in column
+    int64_t RoughMin([[maybe_unused]] Filter *f = nullptr, [[maybe_unused]] common::RSValue *rf = nullptr) override {
```
Comment (RingsC) [inline storage/tianmu/core/temp_table.h]: although your way is right, but we recommend following the conventional usage as below.
```
//innodb handler definitions.
  FT_INFO *ft_init_ext_with_hints(uint inx [[maybe_unused]],
                                  String *key [[maybe_unused]],
                                  Ft_hints *hints [[maybe_unused]]) override {
    ut_d(ut_error);
    ut_o(return (nullptr));
  }

  int ft_read(uchar *buf [[maybe_unused]]) override {
    ut_d(ut_error);
    ut_o(return (HA_ERR_WRONG_COMMAND));
  }
```

**Review by RingsC — CHANGES_REQUESTED** [review]

File: storage/tianmu/core/temp_table.h (hunk 1 above)
Comment (DandreChen) [reply storage/tianmu/core/temp_table.h]: fixed it.  
I will follow this conventional usage.

**Review by duanfuxiang0 — APPROVED** [review]
LGTM

**Review by RingsC — APPROVED** [review]
LGTM

**Pull request merged**